pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -796,3 +803,432 @@ func openMetricsEscape(s string) string {
+}
+
+// TimeInterval describes intervals of time
+type TimeInterval struct {
+	// Times is a list of TimeRange
+	// +optional
+	Times []TimeRange `json:"times,omitempty"`
+	// Weekdays is a list of WeekdayRange
+	// +optional
+	Weekdays []WeekdayRange `json:"weekdays,omitempty"`
+	// DaysOfMonth is a list of DayOfMonthRange
+	// +optional
+	DaysOfMonth []DayOfMonthRange `json:"daysOfMonth,omitempty"`
+	// Months is a list of MonthRange
+	// +optional
+	Months []MonthRange `json:"months,omitempty"`
+	// Years is a list of YearRange
+	// +optional
+	Years []YearRange `json:"years,omitempty"`
+}
+
+// Time defines a time in 24hr format
+// +kubebuilder:validation:Pattern=`^((([01][0-9])|(2[0-3])):[0-5][0-9])$|(^24:00$)`
+type Time string
+
+// TimeRange defines a start and end time in 24hr format
+type TimeRange struct {
+	// StartTime is the start time in 24hr format.
+	StartTime Time `json:"startTime,omitempty"`
+	// EndTime is the end time in 24hr format.
+	EndTime Time `json:"endTime,omitempty"`
+}
+
+// ParsedRange is an integer representation of a range
+type ParsedRange struct {
+	// Start is the beginning of the range
+	Start int `json:"start,omitempty"`
+	// End of the range
+	End int `json:"end,omitempty"`
+}
+
+// WeekdayRange is an inclusive range of days of the week beginning on Sunday
+// Days can be specified by name (e.g 'Sunday') or as an inclusive range (e.g 'Monday:Friday')
+// +kubebuilder:validation:Pattern=`^((?i)sun|mon|tues|wednes|thurs|fri|satur)day(?:((:(sun|mon|tues|wednes|thurs|fri|satur)day)$)|$)`
+type WeekdayRange string
+
+// DayOfMonthRange is an inclusive range of days of the month beginning at 1
+type DayOfMonthRange struct {
+	// Start of the inclusive range
+	// +kubebuilder:validation:Minimum=-31
+	// +kubebuilder:validation:Maximum=31
+	Start int `json:"start,omitempty"`
+	// End of the inclusive range
+	// +kubebuilder:validation:Minimum=-31
+	// +kubebuilder:validation:Maximum=31
+	End int `json:"end,omitempty"`
+}
+
+// MonthRange is an inclusive range of months of the year beginning in January
+// Months can be specified by name (e.g 'January') by numerical month (e.g '1') or as an inclusive range (e.g 'January:March', '1:3', '1:March')
+// +kubebuilder:validation:Pattern=`^((?i)january|febuary|march|april|may|june|july|august|september|october|november|december|[1-12])(?:((:((?i)january|febuary|march|april|may|june|july|august|september|october|november|december|[1-12]))$)|$)`
+type MonthRange string
+
+// YearRange is an inclusive range of years
+// +kubebuilder:validation:Pattern=`^2\d{3}(?::2\d{3}|$)`
+type YearRange string
+
+// Validate the MuteTimeInterval
+func (mti MuteTimeInterval) Validate() error {
+	if mti.Name == "" {
+		return errors.New("empty name field for mute time interval")
+	}
+
+	for i, ti := range mti.TimeIntervals {
+		for _, time := range ti.Times {
+			if err := time.Validate(); err != nil {
+				return fmt.Errorf("time range at %d is invalid - %w", i, err)
```

Comment:
(nit) the idiomatic way of wrapping errors is with a `:`
```suggestion
				return fmt.Errorf("time range at %d is invalid: %w", i, err)
```

and the same applies below